tests/backup_test.php

```diff
@@ -0,0 +1,236 @@
+                        'feedbackformat' => FORMAT_HTML,
+                        'answer' => 'PARIS'
+                    ],
+                    [
+                        'clue' => '<p>Where is the Leaning Tower of Pisa?</p>',
+                        'clueformat' => FORMAT_HTML,
+                        'feedback' => '<p dir="ltr" style="text-align: left;">You are correct.<br></p>',
+                        'feedbackformat' => FORMAT_HTML,
+                        'answer' => 'ITALY'
+                    ]
+                ],
+                'version' => 4,
+            ],
+            'before upgrade feedback column 3.11' => [
+                'filename' => 'crossword_before_feedback_upgrade 3.11.mbz',
+                'coursefullname' => 'before upgrade feedback column 3.11',
+                'courseshortname' => 'bufc311',
+                'questionname' => 'crossword-001',
+                'words' => [
+                    [
+                        'clue' => 'where is the Christ the Redeemer statue located in?',
+                        'clueformat' => FORMAT_HTML,
+                        'feedback' => null,
+                        'feedbackformat' => FORMAT_HTML,
+                        'answer' => 'BRAZIL'
+                    ],
+                    [
+                        'clue' => 'Eiffel Tower is located in?',
+                        'clueformat' => FORMAT_HTML,
+                        'feedback' => null,
+                        'feedbackformat' => FORMAT_HTML,
+                        'answer' => 'PARIS'
+                    ],
+                    [
+                        'clue' => 'Where is the Leaning Tower of Pisa?',
+                        'clueformat' => FORMAT_HTML,
+                        'feedback' => null,
+                        'feedbackformat' => FORMAT_HTML,
+                        'answer' => 'ITALY'
+                    ]
+                ],
+                'version' => 3,
+            ],
+        ];
+    }
+
+    /**
+     * Test old sq backup data from earlier version.
+     *
+     * @covers \restore_qtype_crossword_plugin
+     * @dataProvider test_cw_backup_data_provider
+     * @param string $filename file name of the backup file.
+     * @param string $coursefullname course full name.
+     * @param string $courseshortname course short name
+     * @param string $questionname question name to check after restore.
+     * @param array $expectedwords word data to be checked after restore.
+     * @param int $version skip the test if backup version higher than current major version.
+     */
+    public function test_cw_backup_data(string $filename, string $coursefullname, string $courseshortname,
+        string $questionname, array $expectedwords, int $version): void {
+        global $DB;
+        $this->resetAfterTest();
+        $this->setAdminUser();
+        if (self::get_moodle_version_major() < $version) {
```

Comment:
Why is Moodle version relevant here? I'm sure it was in the StudentQuiz test you copied, but this is not StudentQuiz.

Reply:
Hi @timhunt , the crossword qtype still supports Moodle 3.11, so our backup of 4.0 is not compatible with the 3.11, so I need to make sure, it is tested on the correct version.

Reply:
OK, but then that needs to be a comment in the code, not just a comment in Github which no-one will ever see again.